### PR TITLE
Backend starter code

### LINKED ISSUE #2: Implement backend starter code
- [x] Add Task schema
```
title: string
description: string | null
isChecked: boolean
dateCreated: string (ISO 8601 date format)
```
- [x] Add route: `POST /api/task`
- [x] Add route: `GET /api/task/:id`
- [x] Add route: `DELETE /api/task/:id`

### PR DESCRIPTION
**Pull request by navidboloorian**
## Changes

What changes did you make? Include screenshots if applicable, or explain how to view the changes.

- Added Task schema
- Added get, delete, and post route for Task
- Removed type: module from backend package.json (was a breaking change in previous version)

## Testing

How did you confirm your changes work? (Automated tests, manual verification, etc.)

- Used postman to confirm routes were working properly
- Checked database upon making changes

## Tracking

Add your issue number below.

Resolves #2

### CODE REVIEW

**Comment by navidboloorian** [comment]
Forgive the mess above, I was trying to figure out how to remove the previous commit history but it didn't work... was considering a rebase but didn't end up doing it.

**Comment by wllmwu** [comment]
No worries, it's easy to forget what branch you're on with Git lol

**Review by wllmwu — COMMENTED** [review]
Looks great, just one question about a comment somewhere

File: backend/src/controllers/task.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  try {
+    // if the id doesn't exist, returns null
+    const task = await TaskModel.findById(id);
+
+    if (task === null) {
+      // 404 is the "not found" error
+      throw createHttpError(404, "Task not found.");
+    }
+
+    // if the task exists, it will be the only item within the array
```
[the remark this answers is not in the document]
Comment (navidboloorian) [reply backend/src/controllers/task.ts:21]: Good catch, I had forgotten that findById existed and was just using find. Made the change to findById but forgot to update the comment.

File: backend/src/app.ts [hunk 2]
```diff
@@ -1,14 +1,46 @@
 /**
- * Defines routes.
+ * Defines server and middleware.
  */
 
 import "dotenv/config";
-import express from "express";
+import express, { NextFunction, Request, Response } from "express";
+import taskRoutes from "./routes/task";
```
[the remark this answers is not in the document]
Comment (navidboloorian) [reply backend/src/app.ts:7]: Yeah I'm not sure either... a little confusing.

File: backend/src/app.ts [hunk 3]
```diff
@@ -1,14 +1,46 @@
...
+app.use(express.json());
+
+app.use("/api/task", taskRoutes);
+
+/**
+ * Error handler; all errors thrown by server are handled here.
+ * Explicit typings required here because TypeScript cannot infer the argument types.
+ *
+ * An eslint-disable is being used below because the "next" argument is never used. However,
+ * it is still required for express to recognize it as an error handler. For this reason, I've
```
[the remark this answers is not in the document]
Comment (navidboloorian) [reply backend/src/app.ts:22]: Interesting, I actually wasn't aware that was a thing. Guess we're all learning lol. I'll keep this but add a note for underscores as well.

**Review by wllmwu — APPROVED** [review]
Noice 💯

**Pull request merged**